### PR TITLE
GPTQ export w/a

### PR DESCRIPTION
**Pull request by AlexKoff88**
- These changes are required to export GPTQ models without GPU

### CODE REVIEW

File: optimum/intel/openvino/modeling_decoder.py [hunk 1]
```diff
@@ -227,6 +227,34 @@ def _from_transformers(
...
 
+        # Patch the modules to export of GPTQ models w/o GPU
+        do_gptq_patching = False
+        config_dict = config.to_dict()
+        quantization_config = config_dict.get("quantization_config", None)
+        do_gptq_patching = quantization_config and quantization_config["quant_method"] == "gptq"
+        if do_gptq_patching:
+            torch.set_default_dtype(torch.float32)
+            orig_cuda_check = torch.cuda.is_available
+            torch.cuda.is_available = lambda: True
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_decoder.py:238]: why is this needed ?

File: optimum/intel/openvino/modeling_decoder.py [hunk 2]
```diff
@@ -238,10 +266,14 @@ def _from_transformers(
...
             trust_remote_code=trust_remote_code,
-            model_kwargs=kwargs,
             int8=load_in_8bit,
         )
 
+        # Unpatch modules after GPTQ export
+        if do_gptq_patching:
+            torch.cuda.is_available = orig_cuda_check
+            GPTQQuantizer.post_init_model = orig_post_init_model
+
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_decoder.py:276]: I think it could make sense to add a test as well to verify we don't break anything in the future

**Review by echarlaix — COMMENTED** [review]
Thanks for this addition @AlexKoff88 !

File: optimum/intel/openvino/modeling_decoder.py (hunk 1 above)
Comment (AlexKoff88) [reply optimum/intel/openvino/modeling_decoder.py:238]: GPTQ-optimized models from Transformers require NVidia GPU to be available for inference. We are introducing two changes to overcome this limitation:
- Patching some functions that prevent instantiating AutoModelForCausalLM without GPUs
- Patching QLinear forward in AutoGPTQ to enable model tracing via TorchScript

The first part comes to Optimum while the second is to OpenVINO PyTorch Frontend.

File: optimum/intel/openvino/modeling_decoder.py (hunk 2 above)
Comment (AlexKoff88) [reply optimum/intel/openvino/modeling_decoder.py:276]: I can add a test that checks that AutoModelForCausalLM can be instantiated without GPU. What do you think?

File: optimum/intel/openvino/modeling_decoder.py (hunk 1 above)
Comment (echarlaix) [reply optimum/intel/openvino/modeling_decoder.py:238]: thanks for the explanation !

File: optimum/intel/openvino/modeling_decoder.py (hunk 2 above)
Comment (echarlaix) [reply optimum/intel/openvino/modeling_decoder.py:276]: We can add a test for GPTQ quantized model export + inference once the PR is merged in AutoGPTQ then

Comment (echarlaix) [reply optimum/intel/openvino/modeling_decoder.py:276]: My bad, it was clearly explained in your other comment. Let's merge, we can add a test once enabled in the next openvino release
